lib/ex_doc/formatter/html.ex

```diff
@@ -384,10 +386,31 @@ defmodule ExDoc.Formatter.HTML do
     end)
   end
 
+  defp normalize_extras(base) when is_binary(base), do: {base, %{}}
+  defp normalize_extras({base, opts}), do: {base, Map.new(opts)}
+
   defp disambiguate_id(extra, discriminator) do
     Map.put(extra, :id, "#{extra.id}-#{discriminator}")
   end
 
+  defp build_extra({input, %{url: _} = input_options}, groups, _lang, _auto, _url_pattern) do
+    input = to_string(input)
+    title = input_options[:title] || filename_to_title(input)
+    group = GroupMatcher.match_extra(groups, input)
+
+    # TODO: Can we make the content/source a link?
+
+    %{
+      group: group,
+      id: Utils.text_to_id(title),
+      source_path: input_options[:url],
+      source_url: input_options[:url],
```

Comment:
These probably aren't necessary now.

Reply:
Yes, we probably don't need `id`, `source_path`, `source_url` and `title_content`, as those are all about the rendered page.